archive/logic_puzzles/knights_and_knaves.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2020 Dan Stanescu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: D. Stanescu.
+-/
+
+import tactic
+
+/-!
+# Knights and knaves puzzles
+Two puzzles from "Knights and Knaves" by Raymond Smullyan.
+
+There is an island where all inhabitants are either knights or knaves. 
+Knights always tell the truth. Knaves always lie.
+Jack and Bob are two inhabitants of this island.
```

Comment:
Not a complaint about this PR, but I wonder if you could build the "knights tell the truth; knaves lie" into the actual definitions, rather than stating it as a hypothesis in each problem.

Reply:
That is built-in actually.